src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -970,4 +975,11 @@ public static void onEmptyClick(EntityPlayer player, EnumHand hand)
     {
         MinecraftForge.EVENT_BUS.post(new PlayerInteractEvent.RightClickEmpty(player, hand));
     }
+    
+    public static ITradeList[] onVillagerTradePopulation(EntityVillager villager, int forgeCareerID, ITradeList[] originalTrades)
+    {
+        VillagerTradePopulationEvent event = new VillagerTradePopulationEvent(villager, villager.getProfessionForge().getCareer(forgeCareerID), Lists.newArrayList(originalTrades));
+        MinecraftForge.EVENT_BUS.post(event);
+        return event.getTrades().toArray(new ITradeList[]{});
```

Comment:
Do not call `toArray` with zero length array because it will cause array allocation twice
For the perfomance improvements you can do it like this
```
event.getTrades().toArray(new ITradeList[event.getTrades().size()])
```